Content.Server/Ninja/Systems/NinjaSuitSystem.cs

```diff
@@ -76,6 +87,16 @@ private void OnSuitInsertAttempt(EntityUid uid, NinjaSuitComponent comp, Contain
         RaiseLocalEvent(user, ref ev);
     }
 
+    // this function assigns a score to a power cell depending on the capacity, to be used when comparing which cell is better.
+    private float GetCellScore(EntityUid? uid, BatteryComponent battcomp)
+    {
+        // if a cell is able to automatically recharge, boost the score drastically depending on the recharge rate,
+        // this is to ensure a ninja can still upgrade to a micro reactor cell even if they already have a medium or high.
+        if (TryComp<BatterySelfRechargerComponent>(uid, out var selfcomp) && selfcomp.AutoRecharge)
+            return battcomp.MaxCharge + (selfcomp.AutoRechargeRate*100);
+        return battcomp.MaxCharge;
+    }
```

Comment:
This should really be Entity<BatteryComponent> or make the EntityUid non-nullable.

Also don't put magic numbers in methods, put it as a constant on the system or make it a cvar instead and name it.